### PR TITLE
991 Edit Docket Record

### PR DESCRIPTION
**Pull request by wbyoko**
Screenshot 2019-10-10 15 06 21
Screenshot 2019-10-10 15 06 27

### CODE REVIEW

File: web-client/src/presenter/actions/EditDocketRecord/deconstructReceivedAtDateToFormAction.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+}) => {
+  const { docketEntry } = props;
+
+  const receivedAt = applicationContext
+    .getUtilities()
+    .prepareDateFromString(docketEntry.receivedAt);
+  if (
+    receivedAt &&
+    receivedAt.toDate() instanceof Date &&
+    !isNaN(receivedAt.toDate())
```
Comment (sutt0n) [inline web-client/src/presenter/actions/EditDocketRecord/deconstructReceivedAtDateToFormAction.js]: could we use `receivedAt.isValid()` here?

File: web-client/src/presenter/actions/EditDocketRecord/deconstructReceivedAtDateToFormAction.test.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import { applicationContext } from '../../../applicationContext';
+import { deconstructReceivedAtDateToFormAction } from './deconstructReceivedAtDateToFormAction';
+import { prepareDateFromString } from '../../../../../shared/src/business/utilities/DateHandler';
+import { presenter } from '../../presenter';
+import { runAction } from 'cerebral/test';
+
+presenter.providers.applicationContext = applicationContext;
+
+describe('deconstructReceivedAtDateToFormAction', () => {
+  it('decontructs the date', async () => {
```
Comment (sutt0n) [inline web-client/src/presenter/actions/EditDocketRecord/deconstructReceivedAtDateToFormAction.test.js]: typo -- `deconstructs`

File: web-client/src/presenter/actions/EditDocketRecord/deconstructReceivedAtDateToFormAction.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { state } from 'cerebral';
+
+/**
+ * sets the form's irs notice date and pay gov date based on the caseDetail provided in state.caseDetail
+ *
+ * @param {object} providers the providers object
+ * @param {Function} providers.get the cerebral get function used for getting the state.caseDetail
```
Comment (sutt0n) [inline web-client/src/presenter/actions/EditDocketRecord/deconstructReceivedAtDateToFormAction.js]: should this be `providers.props`?

File: web-client/src/presenter/actions/EditDocketRecord/setDocketEntryFormForDocketEditAction.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import { omit } from 'lodash';
+import { state } from 'cerebral';
+
+/**
+ * sets the current docket entry data for edit
+ *
+ * @param {object} providers the providers object
+ * @param {object} providers.get the cerebral get method
+ * @param {object} providers.set the cerebral set method
```
Comment (sutt0n) [inline web-client/src/presenter/actions/EditDocketRecord/setDocketEntryFormForDocketEditAction.js]: `providers.props`?

File: web-client/src/presenter/sequences/gotoCompleteDocketEntrySequence.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+import { set } from 'cerebral/factories';
+import { setCaseAction } from '../actions/setCaseAction';
+import { setCurrentPageAction } from '../actions/setCurrentPageAction';
+import { setDocketEntryFormForEditAction } from '../actions/EditDocketRecord/setDocketEntryFormForEditAction';
+import { setDocumentIdAction } from '../actions/setDocumentIdAction';
+import { state } from 'cerebral';
+
+export const gotoEditDocketEntry = [
+  setCurrentPageAction('Interstitial'),
+  set(state.showValidation, false),
```
Comment (sutt0n) [inline web-client/src/presenter/sequences/gotoCompleteDocketEntrySequence.js]: i think we have an action that does this, `stopShowValidationAction`

File: web-client/src/presenter/sequences/gotoCompleteDocketEntrySequence.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  clearScansAction,
+  clearFormAction,
+  clearScreenMetadataAction,
+  getCaseAction,
+  setCaseAction,
+  setDocketEntryFormForEditAction,
+  setDocumentIdAction,
+  set(state.isEditingDocketEntry, true),
+  set(state.wizardStep, 'PrimaryDocumentForm'),
+  set(state.documentUploadMode, 'scan'),
```
Comment (sutt0n) [inline web-client/src/presenter/sequences/gotoCompleteDocketEntrySequence.js:27]: could we use `setDocumentUploadModeAction` here and pass the same key in `props`, perhaps?

File: web-client/src/views/EditDocketEntry/PrimaryDocumentForm.jsx [hunk 7]
```diff
@@ -0,0 +1,474 @@
...
+    caseDetail,
+    form,
+    internalTypesHelper,
+    updateDocketEntryFormValueSequence,
+    updateScreenMetadataSequence,
+    validateDocketEntrySequence,
+    validationErrors,
+  }) => {
+    return (
+      <React.Fragment>
```
Comment (sutt0n) [inline web-client/src/views/EditDocketEntry/PrimaryDocumentForm.jsx]: for consistency's sake, `<>` could go here and its corresponding `</>` at the bottom

**Review by sutt0n — CHANGES_REQUESTED** [review]
Just a few things here and there.

File: web-client/src/presenter/sequences/gotoCompleteDocketEntrySequence.js (hunk 6 above)
Comment (wbyoko) [reply web-client/src/presenter/sequences/gotoCompleteDocketEntrySequence.js:27]: this is a default coming from a url. I doubt we want to default props in the router.

**Review by sutt0n — APPROVED** [review]

**Pull request merged**